mbed_tools_ci_scripts/utils/git_helpers.py

```diff
@@ -36,6 +36,11 @@ def __init__(self, path: Path, repo: Repo) -> None:
             configuration.get_value(ConfigurationVariable.BOT_EMAIL),
         )
 
+    def __del__(self) -> None:
```

Comment:
We don't claim to support python 3.5, do we need this fix for 3.6 or above?

Reply:
Also I would avoid using `__del__`, using it can break garbage collection in a number of ways. We'd be better off making this a context manager and calling `repo.close` in `__exit__`, as then at least the method is guaranteed to be called at the end of a `with` block.

Reply:
> We don't claim to support python 3.5, do we need this fix for 3.6 or above?

Yeah CI failure is happening with Windows and python 3.6 not 3.5 (incorrect commit message).

Reply:
> Also I would avoid using `__del__`, using it can break garbage collection in a number of ways. We'd be better off making this a context manager and calling `repo.close` in `__exit__`, as then at least the method is guaranteed to be called at the end of a `with` block.

Ah k. Let me check that.

Reply:
The changes I did are based on what is done in gitpython https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L217. `gc.collect()` is invoked in `close()`. Is it ok to call garbage collection in `__exit__ `?

Reply:
Oh my word, that GitPython code is an awful hack. So many things wrong there. Anyway, we should be OK to call the `close` method in `__exit__`, they do it themselves https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L214 -- the garbage collector should only clean up any objects whose reference count has reached 0 (although there are other subtleties like generational garbage collection). I'm interested in what the test failure actually is and why we weren't seeing it before.

Reply:
CI failure only happens with Windows and python 3.6 https://dev.azure.com/mbed-tools/mbed-tools-ci-scripts/_build/results?buildId=3277&view=ms.vss-test-web.build-test-results-tab
`git.exc.GitCommandNotFound: Cmd('git') not found due to: OSError('[WinError 6] The handle is invalid')` is one of the errros.